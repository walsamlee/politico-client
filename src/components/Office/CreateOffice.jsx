/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';

class CreateOffice extends Component {
  constructor(props) {
    super(props);

    this.state = {
      officeid: '',
      officetype: '',
      officename: '',
    };
  }

  submitHandler = (event) => {
    event.preventDefault();
    const token = localStorage.getItem('token');

    if (token) {
      const officeData = {
        id: this.state.officeid,
        type: this.state.officetype,
        name: this.state.officename,
      };

      this.props.createOffice(officeData, token);
    }
  }

  resetHandler = () => {
    this.setState({
      officeid: '',
      officetype: '',
      officename: '',
    });
  };

  changeHandler = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <div className="main">
        <div className="dashboard">
          <div className="signup-form">
            <form id="create-office" name="sign-up" onSubmit={this.submitHandler} onReset={this.resetHandler}>
              <table className="form-table">
                <tr>
                  <td className="label" />
                  <td>
                    <input id="office-id" type="text" name="officeid" placeholder="Office ID" required onChange={this.changeHandler} value={this.state.officeid} />
                  </td>
                </tr>
                <tr>
                  <td className="label" />
                  <td>
                    <input id="office-type" type="text" name="officetype" placeholder="Office Type" required onChange={this.changeHandler} value={this.state.officetype} />
                  </td>
                </tr>
                <tr>
                  <td className="label" />
                  <td>
                    <input id="officename" type="text" name="officename" placeholder="Office Name" required onChange={this.changeHandler} value={this.state.officename} />
                  </td>
                </tr>
                <tr>
                  <td />
                  <td>
                    <input type="submit" className="btn btn-cart" value="Save" />
                    <input type="reset" className="btn btn-cart" value="Reset" onClick={this.resetHandler} />
                  </td>
                </tr>
              </table>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default CreateOffice;
