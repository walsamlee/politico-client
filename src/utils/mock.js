const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
