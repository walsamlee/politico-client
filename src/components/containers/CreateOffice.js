import { connect } from 'react-redux';
import CreateOffice from '../Office/CreateOffice';
import officeAction from '../../actions/officeAction';

const mapStateToProps = state => ({
  data: state,
});

const mapDispatchToProps = dispatch => ({
  createOffice: (officeData, token) => {
    dispatch(officeAction.createOffice(officeData, token));
  },
});

const createNewOffice = connect(mapStateToProps, mapDispatchToProps)(CreateOffice);

export default createNewOffice;
