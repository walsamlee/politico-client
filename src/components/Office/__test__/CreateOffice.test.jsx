import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import CreateOffice from '../CreateOffice';

const { token } = mock;

describe('<CreateOffice />', () => {
  it('should render CreateOffice component', () => {
    const wrapper = shallow(
      <CreateOffice token={token} createOffice={jest.fn()} />,
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: '',
        value: '',
      },
    };
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
    wrapper.instance().submitHandler(event);
    wrapper.instance().changeHandler(event);
    wrapper.instance().resetHandler();
  });
});
