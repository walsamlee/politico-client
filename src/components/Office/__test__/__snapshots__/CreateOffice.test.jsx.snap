// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateOffice /> should render CreateOffice component 1`] = `ShallowWrapper {}`;
